add_to_app/android_view/app/src/main/java/dev/flutter/example/androidView/ListAdapter.kt

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package dev.flutter.example.androidView
+
+import android.content.Context
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import dev.flutter.example.androidView.databinding.AndroidCardBinding
+import io.flutter.embedding.android.FlutterView
+import io.flutter.plugin.common.MethodChannel
+import java.util.*
+import kotlin.random.Random
+
+class ListAdapter(private val context: Context, private val flutterViewEngine: FlutterViewEngine) : RecyclerView.Adapter<ListAdapter.Cell>() {
+    var previousFlutterCells = TreeSet<Int>();
+
+    private val matchParentLayout = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+
+    private val random = Random.Default
+    private val flutterView = FlutterView(context)
+    private val flutterChannel = MethodChannel(flutterViewEngine.engine.dartExecutor, "dev.flutter.example/cell")
+
+    private var flutterCell: Cell? = null
+
+    inner class Cell(val binding: AndroidCardBinding) : RecyclerView.ViewHolder(binding.root) {
+
+    }
+
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): Cell {
+        val binding = AndroidCardBinding.inflate(LayoutInflater.from(viewGroup.context), viewGroup, false)
+
+        // Let the default view holder have an Android card. When needed, hide the Android card
+        // and show a Flutter one instead.
+        return Cell(binding)
+    }
+
+    override fun onBindViewHolder(cell: Cell, position: Int) {
+        // While scrolling forward, if no Flutter is presently showing, let the next one have a 1/3
+        // chance of being Flutter.
+        //
+        // While scrolling backward, let it be deterministic, and only show cells that were
+        // previously Flutter cells as Flutter cells.
+        if (previousFlutterCells.contains(position)
+            || ((previousFlutterCells.isEmpty() || position > previousFlutterCells.last())
+                && flutterCell == null
+                && random.nextInt(3) == 0)) {
```

Comment:
You could be more deterministic here by using the position with a prime number: `(position * prime) % 3 == 0`

Reply:
ya, it's a bit arbitrary, but I'm attempting to add a bit of immersion too to make it seem a bit like it could be server directed and not entirely pre-scripted.